public/app/features/dashboard/utils/getPanelMenu.test.ts

```diff
@@ -94,117 +95,285 @@ describe('getPanelMenu', () => {
     `);
   });
 
-  describe('when panel is in view mode', () => {
-    it('should return the correct panel menu items', () => {
-      const getExtendedMenu = () => [{ text: 'Toggle legend', shortcut: 'p l', click: jest.fn() }];
-      const ctrl: any = { getExtendedMenu };
-      const scope: any = { $$childHead: { ctrl } };
-      const angularComponent: any = { getScope: () => scope };
-      const panel = new PanelModel({ isViewing: true });
-      const dashboard = createDashboardModelFixture({});
-
-      const menuItems = getPanelMenu(dashboard, panel, angularComponent);
-      expect(menuItems).toMatchInlineSnapshot(`
-        [
-          {
-            "iconClassName": "eye",
-            "onClick": [Function],
-            "shortcut": "v",
-            "text": "View",
-          },
-          {
-            "iconClassName": "edit",
-            "onClick": [Function],
-            "shortcut": "e",
-            "text": "Edit",
-          },
-          {
-            "iconClassName": "share-alt",
-            "onClick": [Function],
-            "shortcut": "p s",
-            "text": "Share",
-          },
-          {
-            "iconClassName": "compass",
-            "onClick": [Function],
-            "shortcut": "x",
-            "text": "Explore",
-          },
-          {
-            "iconClassName": "info-circle",
-            "onClick": [Function],
-            "shortcut": "i",
-            "subMenu": [
-              {
-                "onClick": [Function],
-                "text": "Panel JSON",
-              },
-            ],
-            "text": "Inspect",
-            "type": "submenu",
-          },
-          {
-            "iconClassName": "cube",
-            "onClick": [Function],
-            "subMenu": [
-              {
-                "href": undefined,
-                "onClick": [Function],
-                "shortcut": "p l",
-                "text": "Toggle legend",
-              },
-            ],
-            "text": "More...",
-            "type": "submenu",
-          },
-        ]
-      `);
-    });
+  it('should return the correct panel menu items when data is streaming', () => {
+    const panel = new PanelModel({});
+    const dashboard = createDashboardModelFixture({});
+
+    const menuItems = getPanelMenu(dashboard, panel, LoadingState.Streaming);
+    expect(menuItems).toMatchInlineSnapshot(`
```

Comment:
if streaming is the thing you want to test here, better to check that items exists rather than big snapshot. This can fail for many reasons not related to loading state (IE use snapshot tests sparingly)

Reply:
Yes, snapshots are expensive.. good catch, thanks! Fixed 🙏